### ISSUE TITLE
Incorrect false positive CS0162, IDE0035 when constant referenced by local method

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#41429]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/906120/incorrect-false-positive-cs0162-ide0035-when-const.html)._

---
```csharp
using System;

class Program
{
    static void Main(string[] args)
    {
        Console.WriteLine("Hello World! " + checkSomething(4));
        return;

const int cutoff = 3; // CS0162, IDE0035 on this line

static bool checkSomething(int value) => value < cutoff;
    }
}
```

I do not expect an unreachable code warning or the unused variable warning. Deleting that line breaks the compilation.

---
#### Visual Studio Feedback System on 2/4/2020, 07:18 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by CyrusNajmabadi** [comment]
Major issue here seems to be the compiler warning.  That just seems incorrect.

**Comment by binki** [comment]
Furthermore, `const` is special: it’s not something that is ever executed, so it’s never even “reached” in one sense.

**Comment by manfred-brands** [comment]
This is still happening in VS2022 with .NET6 RC2.

**Comment by shuebner** [comment]
Still an issue with at least SDK 6.0.405
